presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -193,16 +195,17 @@ public String toText(boolean verbose, int level)
         return new TextRenderer(verbose, level).render(representation);
     }
 
-    public String toJson()
+    public String toJson(FunctionAndTypeManager type)
     {
-        return new JsonRenderer().render(representation);
+        this.functionAndTypeManager = type;
+        return new JsonRenderer(type).render(representation);
```

Comment:
Why can't we use the `functionAndTypeManager` that we instantiate with `PlanPrinter`

Reply:
no you are right, this pass-in is not needed